File: backend/src/main/java/outcobra/server/model/Teacher.java
package outcobra.server.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.ManyToOne;
import javax.validation.constraints.NotNull;
import java.util.List;

/**
 * Created by Florian on 22.10.2016.
 */

@Data
@Entity
@AllArgsConstructor
@NoArgsConstructor
public class Teacher {

    @Id
    private long id;

    @ManyToOne
    private Institution institution;


    @NotNull
    private String name;

    private String email;

}

File: backend/src/main/kotlin/outcobra/server/model/domain/Teacher.kt
package outcobra.server.model.domain

import outcobra.server.model.interfaces.ParentLinked
import javax.persistence.Entity
import javax.persistence.ManyToOne
import javax.persistence.OneToMany
import javax.validation.constraints.NotNull

@Entity
data class Teacher(@NotNull var name: String = "",
                   var email: String? = null,
                   @ManyToOne var institution: Institution? = null,
                   @OneToMany(mappedBy = "teacher") var subjects: List<Subject> = listOf())
    : ParentLinked, AbstractEntity() {

    override val parent: ParentLinked?
        get() = institution
}